app-services/cloud-event-transforms/pkg/transforms/conversion.go

```diff
@@ -0,0 +1,128 @@
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+// Conversion houses various conversion
+type Conversion struct {
+}
+
+// NewConversion creates, initializes and returns a new instance of Conversion
+func NewConversion() Conversion {
+	return Conversion{}
+}
+
+// TransformToCloudEvent will transform a models.Event to a Cloud Event
+// It will return an error and stop the pipeline if a non-edgex event is received or if no data is received.
+func (f Conversion) TransformToCloudEvent(edgexcontext *appcontext.Context, params ...interface{}) (continuePipeline bool, stringType interface{}) {
+	if len(params) < 1 {
+		return false, errors.New("No Event Received")
+	}
+	edgexcontext.LoggingClient.Debug("Transforming to CloudEvent")
+	event, ok := params[0].(models.Event)
+	if !ok {
+		return false, errors.New("Unexpected type received")
+	}
+	if len(event.Readings) == 0 {
+		return false, errors.New("No event readings to transform")
+	}
+
+	var cloudeventReadings []cloudevents.Event
+	for _, reading := range event.Readings {
+		cloudevent := cloudevents.NewEvent(cloudevents.VersionV1)
+		cloudevent.SetID(reading.Id)
+		cloudevent.SetType(reading.Name)
+		cloudevent.SetSource(event.Device)
+		unixTime := time.Unix(0, reading.Origin) // assuming time is formatted as nanoseconds if seconds time.Unix(reading.Origin, 0)
+		timeRFC3339, err := time.Parse(
+			time.RFC3339,
+			unixTime.Format(time.RFC3339))
+		if err != nil {
+			return false, fmt.Errorf("Failed to parse reading.Origin as RFC3339 time: %d, %s", reading.Origin, err)
+		}
+		cloudevent.SetTime(timeRFC3339)
+		// extension names are always lowercase
+		cloudevent.SetExtension("eventid", event.ID)
+		cloudevent.SetExtension("valuetype", reading.ValueType)
+		cloudevent.SetExtension("floatencoding", reading.FloatEncoding)
+		if len(reading.BinaryValue) > 0 {
+			// if reading.BinaryValue is set that becomes data and reading.Value is ignored
+			if err := cloudevent.SetData(reading.BinaryValue); err != nil {
+				return false, fmt.Errorf("Error setting data field for cloud event, %s", err)
+			}
+		} else {
+			cloudevent.SetDataContentType("application/json")
+			if err := cloudevent.SetData(reading.Value); err != nil {
+				return false, fmt.Errorf("Error setting data field for cloud event, %s", err)
+			}
+		}
+		cloudeventReadings = append(cloudeventReadings, cloudevent)
+	}
+	return true, cloudeventReadings
+}
+
+// TransformFromCloudEvent will transform a Cloud Event to an Edgex models.Event
+// It will return an error and stop the pipeline if a non-edgex event is received or if no data is received.
+func (f Conversion) TransformFromCloudEvent(edgexcontext *appcontext.Context, params ...interface{}) (continuePipeline bool, stringType interface{}) {
+	if len(params) < 1 {
+		return false, errors.New("No Event Received")
+	}
+	edgexcontext.LoggingClient.Debug("Transforming from CloudEvent to models.Event")
+	if result, ok := params[0].([]cloudevents.Event); ok {
```

Comment:
do the opposite to return error earlier if cast is NOT ok so that the whole code logic needs not indent.